### PR TITLE
openapi: adding watch parameter on get & list operations

### PR DESCRIPTION
**Pull request by lukas-bednar**
Related to #582

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Some tests failed

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -494,6 +508,12 @@
        "description": "Should the export be exact. Exact export maintains cluster-specific fields like 'Namespace'",
        "name": "exact",
        "in": "query"
+      },
+      {
+       "type": "boolean",
```
Comment (lukas-bednar) [inline api/openapi-spec/swagger.json]: this one shouldn't be here, I will drop it

File: api/openapi-spec/swagger.json [hunk 2]
```diff
@@ -634,6 +654,12 @@
...
```
Comment (lukas-bednar) [inline api/openapi-spec/swagger.json]: this one shouldn't be here, I will drop it

File: api/openapi-spec/swagger.json [hunk 3]
```diff
@@ -953,6 +986,12 @@
...
```
Comment (lukas-bednar) [inline api/openapi-spec/swagger.json]: this one shouldn't be here, I will drop it

File: api/openapi-spec/swagger.json (hunk 1 above)
Comment (lukas-bednar) [reply api/openapi-spec/swagger.json]: Actually it is not true, K8S support it here as well, so I will keep it and just change name of PR

**Comment by rmohr** [comment]
ok to test

File: pkg/virt-api/rest/kubeproxy.go [hunk 4]
```diff
@@ -151,31 +151,9 @@ func GenericResourceProxy(ws *restful.WebService, ctx context.Context, gvr schem
...
-			To(NotImplementedYet).Writes(objExample).
-			Doc("Watch a "+objKind+" object.").
-			Returns(http.StatusOK, "OK", objExample).
-			Returns(http.StatusNotFound, "Not Found", nil), ws,
-	))
-
 	ws.Route(addGetNamespacedListParams(
 		ws.GET(ResourceBasePath(gvr)).
-			Produces(mime.MIME_JSON, mime.MIME_YAML).
+			Produces(mime.MIME_JSON, mime.MIME_YAML, mime.MIME_JSON_STREAM).
```
Comment (lukas-bednar) [inline pkg/virt-api/rest/kubeproxy.go]: I need to add `mime.MIME_JSON_STREAM` and `watch` parameter to AllNamespaced list as well.

Comment (lukas-bednar) [reply pkg/virt-api/rest/kubeproxy.go]: Done

File: api/openapi-spec/swagger.json [hunk 5]
```diff
@@ -1264,347 +1321,11 @@
...
-      "default": {
-       "description": "OK",
-       "schema": {
-        "$ref": "#/definitions/v1.VirtualMachineList"
-       }
-      }
-     }
-    }
-   },
-   "/apis/kubevirt.io/v1alpha1/watch/migrations": {
```
Comment (rmohr) [inline api/openapi-spec/swagger.json]: When I look here: https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json, I can still see the /watch/ endpoints.

Comment (lukas-bednar) [reply api/openapi-spec/swagger.json]: Oh, I am sorry ... I don't  know on what I was looking at then.

**Comment by rmohr** [comment]
retest this please

**Pull request merged**